src/sdl2/controller.rs

```diff
@@ -128,25 +132,37 @@ impl GameControllerSubsystem {
         }
     }
 
-    /// Load mappings from a file
-    pub fn load_mappings<P: AsRef<Path>>(&self, path: P)
-            -> Result<i32, AddMappingError> {
+    /// Load controller input mappings from a file.
+    pub fn load_mappings<P: AsRef<Path>>(&self, path: P) -> Result<i32, AddMappingError> {
         use self::AddMappingError::*;
 
-        let file = match RWops::from_file(path, "r") {
-            Ok(f) => f,
-            Err(s) => return Err(InvalidFilePath(s))
-        };
+        let rw = RWops::from_file(path, "r").map_err(InvalidFilePath)?;
+        self.load_mappings_from_rw(rw)
+    }
+
+    /// Load controller input mappings from a [`Read`](std::io::Read) object.
+    pub fn load_mappings_from_read<R: io::Read>(
+        &self,
+        read: &mut R,
+    ) -> Result<i32, AddMappingError> {
+        use self::AddMappingError::*;
 
-        let result = unsafe { sys::SDL_GameControllerAddMappingsFromRW(file.raw(), 0) };
+        let mut buffer = Vec::with_capacity(1024);
+        let rw = RWops::from_read(read, &mut buffer).map_err(ReadError)?;
+        self.load_mappings_from_rw(rw)
+    }
+
+    /// Load controller input mappings from an SDL [`RWops`] object.
+    pub fn load_mappings_from_rw<'a>(&self, rw: RWops<'a>) -> Result<i32, AddMappingError> {
```

Comment:
One of the things I wasn't sure about when making this PR was whether this should be `pub`. Thoughts?